packages/libs/src/sdk/services/Solana/programs/BaseSolanaProgramClient.ts

```diff
@@ -90,14 +105,41 @@ export class BaseSolanaProgramClient {
       instructions,
       feePayer,
       recentBlockhash,
-      addressLookupTables = []
+      addressLookupTables = [],
+      priorityFee
     } = await parseParams('buildTransaction', BuildTransactionSchema)(params)
 
     if (!recentBlockhash) {
       const res = await this.connection.getLatestBlockhash()
       recentBlockhash = res.blockhash
     }
 
+    if (priorityFee !== undefined) {
+      if ('microLamports' in priorityFee) {
+        instructions.push(
+          ComputeBudgetProgram.setComputeUnitPrice({
+            microLamports: priorityFee.microLamports
+          })
+        )
+      } else {
+        const res = await this.connection.getRecentPrioritizationFees()
+        const orderedFees = res.map((r) => r.prioritizationFee).sort()
```

Comment:
Is this logic useful as a separate utility that we just call here?

Reply:
possibly, I think we can extract if needed